### DISCUSSION TITLE
Can I color a note title in tree based on attribute?

### DISCUSSION THREAD

**Discussion by jniggemann** [TriliumNext/Trilium#2759] [Q&A] [answered]
I'd like all notes in the tree that have a certain attribute to automatically have their title colored.
Usecase: attribute #fixme shall make all notes in the tree easily recognizable.

I imagine a new CSS note with my desired colorization and attribute `appCss` would do.
Question is how I add the relevant `cssClass` attribute to all notes automatically. Maybe with a startup script?

**Comment by XXXJetfireXXX** [comment]
You can apply an attribute to the whole subtree by making it inheritable:
![image](https://user-images.githubusercontent.com/43646178/160666501-26bccfef-8bb1-4cf1-9b03-14d9c7b084c0.png)
Unless you only want some specific notes in the subtree to get the attribute which would probably require a script.

**Reply by jniggemann** [reply]
> Unless you only want some specific notes in the subtree to get the attribute

Correct, only ones with a certain attribute (ex. #fixme)


>  which would probably require a script.

Yes, I think so, tool

**Comment by jniggemann** [comment, marked as answer]
Done, leaving it here for whoever is interested.

## Solution

We need two notes, one will contain the desired CSS, the other the implementation.

### CSS Note
This note of type `CSS` needs the label `#appCss`, which makes the CSS code available for later use.
```css
span.fancytree-node.fixme .fancytree-title {
    color: #C70039!important;
    font-weight: bold;
    background-color: #FFC300!important;
    }
```

### Implementation
This note of type `JS frontend` needs the label `#run=frontendStartup`, which makes it execute, well, on frontend startup.  
It searches for all notes having the label `#fixme` and uses `setLabel()` which, in turn, creates `#cssClass=fixme` in the note if it doesn't already exist.  My first idea was running directly on the backend, but I couldn't quickly figure out how to reload the frontend from there (`location.reload()` gave me errors). Using note type `JS frontend` and executing backend code from there *just worked*.
```javascript
api.runOnBackend(() => {
    const myNotes = api.searchForNotes('#fixme');
    for (var i = 0; i < myNotes.length; ++i) {
        myNotes[i].setLabel('cssClass','fixme');
    }   
});
```
